lib/plucky/criteria_hash.rb

```diff
@@ -164,14 +166,12 @@ def modifier_merge(hash, value)
 
     # Private
     def merge_values_into_array(value, other_value)
-      array(value).concat(array(other_value)).uniq
+      array(value).concat(array(other_value)).flatten.uniq
     end
 
-    # Private: Array(BSON::ObjectId) returns the byte array or what not instead
-    # of the object id. This makes sure it is an array of object ids, not the
-    # guts of the object id.
```

Comment:
The imperious: array(nil) returns [] instead of [nil].  I'd like to find a better solution to both of these cases, but I'm not sure what it is.

> On Feb 4, 2016, at 12:46 PM, Chris Heald <notifications@github.com> wrote:
> 
> In lib/plucky/criteria_hash.rb:
> 
> >      end
> >  
> > -    # Private: Array(BSON::ObjectId) returns the byte array or what not instead
> > -    # of the object id. This makes sure it is an array of object ids, not the
> > -    # guts of the object id.
> Why remove the comment here? This is useful for explaining why it's done rather than just returning Array(value) IMO. The change in implementation actually changes the behavior, too; Kernel#Array calls to_ary and to_a on the argument, whereas this just wraps it in a new array. What's the impetus here?
> 
> A quick example of where this may fail is if value is a Set:
> 
> > v = Set.new([1,2,3])
>  => #<Set: {1, 2, 3}>
> > Array(v)
>  => [1, 2, 3]
> > [v]
>  => [#<Set: {1, 2, 3}>]
> —
> Reply to this email directly or view it on GitHub.
>